packages/web3-eth/src/rpc_method_wrappers.ts

```diff
@@ -1206,14 +1201,10 @@ export function sendTransaction<
 							);
 						}
 					} catch (error) {
-						if (error instanceof ContractExecutionError) {
-							promiEvent.emit('contractExecutionError', error);
-						}
-						if (promiEvent.listenerCount('error') > 0) {
-							promiEvent.emit(
-								'error',
-								new TransactionError((error as Error).message),
-							);
+						if (error instanceof InvalidResponseError &&
+							promiEvent.listenerCount('error') > 0
+						) {
+							promiEvent.emit('error', error);
```

Comment:
When this code executes, `error` is one of two things:

1. It's an instance of `InvalidResponseError` (the error will be renamed to something like `RpcError` (#5848)), meaning the transaction failed to send, but it was rejected by the RPC provider and didn't even make it to the Ethereum network. This happens for errors like insufficient funds or unknown account

2. Literally every other possible error that could occur, meaning it could be an error that we haven't anticipated or just an error resulting from a bug we have in the code. In these instances, I don't think it makes sense to cast this error as `TransactionError` because it implies there's an error in the transaction the user is trying to send, but this error could be anything. If we don't cast it though then it makes `error` of type `unknown` and we'd have to include `unknown` as a part of the possible emission types for [SendTransactionEvents](https://github.com/web3/web3.js/blob/36c9b0d950f1b882dcbb28839bda5cd83a9ae5bb/packages/web3-eth/src/types.ts#L42) which makes the entire type show as `unknown` to the user reducing type safety. So, I'm suggesting we only emit an error for expected errors i.e. anything that extends `InvalidResponseError` as all other expected transaction related errors will be handled within the `try` block as the result of the transaction actually being included in a block on chain